java/src/main/java/com/plaid/quickstart/resources/AccessTokenResource.java

```diff
@@ -0,0 +1,68 @@
+package com.plaid.quickstart.resources;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.plaid.client.PlaidClient;
+import com.plaid.client.request.ItemPublicTokenExchangeRequest;
+import com.plaid.client.response.ItemPublicTokenExchangeResponse;
+import com.plaid.quickstart.QuickstartApplication;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import retrofit2.Response;
+
+@Path("/")
+public class AccessTokenResource {
+  static final Logger LOG = LoggerFactory.getLogger(AccessTokenResource.class);
+
+  private PlaidClient plaidClient;
+
+  public AccessTokenResource(PlaidClient _plaidClient) {
+    plaidClient = _plaidClient;
+  }
+
+  @POST
+  @Path("get_access_token")
+  @Produces("application/json")
+  public Object getAccessToken(@FormParam("public_token") String publicToken) throws IOException {
+    Response<ItemPublicTokenExchangeResponse> itemResponse = plaidClient.service()
+        .itemPublicTokenExchange(new ItemPublicTokenExchangeRequest(publicToken))
+        .execute();
+
+    QuickstartApplication.accessToken = itemResponse.body().getAccessToken();
```

Comment:
can we not have the client pass this back in?

Reply:
the `accessToken` needs to be retrieved from `itemResponse.body()`, so I'm not sure how to pass this back to the client without the global state

Reply:
can you not return it in the body of this endpoint and have the client pass it back in.

or alternative store it somewhere like a sqlite database

Reply:
i'm assuming that retrieving the accessToken would only be called once (through Link). would injecting the configuration into the resources and/or locking the variable work?

i don't think returning is possible here, and writing to sqlite/files are probably too unwieldy for this quickstart guide.

Reply:
fair. and this is roughly what we do for ruby/python.

add a comment around the setter to say 'ideally we would store this somewhere more persistent?'

Reply:
done